dotnet/src/SemanticKernel.Core/PromptTemplate/KernelPromptTemplate.cs

```diff
@@ -187,5 +188,13 @@ private static bool ShouldEncodeTags(bool disableTagEncoding, HashSet<string> sa
         return !disableTagEncoding && block is not TextBlock;
     }
 
+    private static string? ConvertToString(object? value, CultureInfo? culture = null)
+    {
+        if (value is null) { return null; }
+
+        return value is IEnumerable<string> enumerableString
+            ? string.Join(" ", enumerableString)
```

Comment:
Didn't we say that we should only convert if it's a known concrete type, because others may have implemented their own tostring logic.
So:
```suggestion
        return value.GetType == typeof(List<string>)
            ? string.Join(" ", (List<string>)value)
```
We also cannot use is since it will also match derived types.